ui/core/components/suggest_reforges_action.tsx

```diff
@@ -1202,12 +1215,19 @@ export class ReforgeOptimizer {
 
 			// Go down the list and include all gems until we find the highest EP option with zero capped stats.
 			const includedGemDataForColor = new Array<GemData>();
+			let foundUncappedJCGem = false;
 
 			for (const gemData of filteredGemDataForColor) {
-				includedGemDataForColor.push(gemData);
+				if (gemData.gem.requiredProfession == Profession.ProfessionUnknown || !foundUncappedJCGem) {
+					includedGemDataForColor.push(gemData);
+				}
 
 				if (!ReforgeOptimizer.includesCappedStat(gemData.coefficients, reforgeCaps, reforgeSoftCaps) && socketColor != GemColor.GemColorCogwheel) {
-					break;
+					if (gemData.gem.requiredProfession > Profession.ProfessionUnknown) {
+						foundUncappedJCGem = true;
+					} else {
+						break;
+					}
 				}
```

Comment:
Given the number of times we are doing JC checks like this, I think we should add another field to the `GemData` type for `isJc : boolean`, which we populate during `buildGemOptions()`.